forte/processors/data_augment/algorithms/skeleton_op.py

```diff
@@ -0,0 +1,765 @@
+                    is_inclusive = (
+                        entry_to_copy != self.configs["augment_entry"]
+                    )
+                    span_new_begin = self.modify_index(
+                        orig_anno.begin, spans, new_spans, True, is_inclusive
+                    )
+                    span_new_end = self.modify_index(
+                        orig_anno.end, spans, new_spans, False, is_inclusive
+                    )
+
+                new_anno = create_class_with_kwargs(
+                    entry_to_copy,
+                    {
+                        "pack": new_pack,
+                        "begin": span_new_begin,
+                        "end": span_new_end,
+                    },
+                )
+                new_pack.add_entry(new_anno)
+                entry_map[orig_anno.tid] = new_anno.tid
+
+            # Deal with spans after the last annotation in the original pack.
+            if entry_to_copy == self.configs["augment_entry"]:
+                while insert_ind < len(inserted_annos):
+                    _insert_new_span(
+                        entry_to_copy,
+                        insert_ind,
+                        inserted_annos,
+                        new_pack,
+                        spans,
+                        new_spans,
+                    )
+                    insert_ind += 1
+
+        # Iterate over and copy the links/groups in the datapack.
+        for link in data_pack.get(Link):
+            self._copy_link_or_group(link, entry_map, new_pack)
+        for group in data_pack.get(Group):
+            self._copy_link_or_group(group, entry_map, new_pack)
+
+        self._data_pack_map[pid] = new_pack.pack_id
+        self._entry_maps[pid] = entry_map
+        return new_pack
+
+    def _overlap_with_existing(self, pid: int, begin: int, end: int) -> bool:
+        r"""
+        This function will check whether the new span
+        has an overlap with any existing spans.
+
+        Args:
+            pid: Datapack Id.
+            begin: The span begin index.
+            end: The span end index.
+        Returns:
+            True if the input span overlaps with
+            any existing spans, False otherwise.
+        """
+        if len(self._replaced_annos[pid]) == 0:
+            return False
+        ind: int = (
+            bisect_left(self._replaced_annos[pid], (Span(begin, begin), "")) - 1
+        )
+        if ind < 0:
+            ind += 1
+
+        while ind < len(self._replaced_annos[pid]):
+            span: Span = self._replaced_annos[pid][ind][0]
+            if not (span.begin >= end or span.end <= begin):
+                return True
+            if span.begin > end:
+                break
+            ind += 1
+
+        return False
+
+    def insert_annotations(
+        self, inserted_text: str, data_pack: DataPack, pos: int
+    ) -> bool:
+        r"""
+        This is a utility function to insert a new annotation. After
```

Comment:
Feels like this is not accurate `This is a utility function to insert a new annotation`. The function only insert some new text right?